README.md

```diff
@@ -473,22 +485,193 @@ When the scanner successfully decodes a `TransactionRequest` it will call its de
+* `onTransactionConsumptionFinalizedFail(APIError)`: Invoked if a `TransactionConsumption` fails to consume the request.
+        
+### TransactionConsumption Event
+ 
+Similarly to the `TransactionRequestCallback`, a `TransactionConsumption` can be listened for incoming confirmations using the `TransactionConsumptionCallback`.
+The possible events are:
+* `onTransactionConsumptionFinalizedSuccess(TransactionConsumption)`: Invoked if a `TransactionConsumption` has been finalized successfully, and the transfer was made between the 2 addresses.
+* `onTransactionConsumptionFinalizedFail(APIError)`: Invoked if a `TransactionConsumption` fails to consume the request.
+
+**Usage**
+```kotlin
+// The transaction requestor listen for the event 
+transactionRequest.startListeningEvents(socketClient, object: SocketCustomEventCallback.TransactionRequestCallback() {
+   override fun onTransactionConsumptionRequest(transactionConsumption: TransactionConsumption) {
+       // Do something
+   }
+
+   override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+       // Do something
+   }
+
+   override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
+       // Do something
+   }
+})
+
+// The transaction consumer listen for the event
+transactionConsumption.startListeningEvents(socketClient, object: SocketCustomEventCallback.TransactionConsumptionCallback() {
+  override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+      // Do something
+  }
+
+  override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
+      // Do something
+  }
+})
+```
+
+> Note: You might want to listen for the event later after you got the `TransactionRequest` or `TransactionConsumption` object, but you might not want to pass the object around.
+In this case, it is possible to keep only the `socket_topic` string directly. Then you can alternatively listening for custom events by using `joinChannel` method of the `OMGSocketClient` instance.
+This implementation will provide the exactly same result as the implementation above ☝️. For example,
+
+```kotlin
+// The transaction requestor listen for the event 
+socketClient.joinChannel(SocketTopic(transactionRequestTopic), listener = object: SocketCustomEventCallback.TransactionRequestCallback(){
+  override fun onTransactionConsumptionRequest(transactionConsumption: TransactionConsumption) {
+      // Do something
+  }
+
+  override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+      // Do something
+  }
+
+  override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
+      // Do something
+  }
+})
+
+// The transaction consumer listen for the event
+socketClient.joinChannel(SocketTopic(transactionRequestTopic), listener = object: SocketCustomEventCallback.TransactionConsumptionCallback() {
```

Comment:
Are the "topics" stored somewhere in a class?
Might be useful to use an enum instead to ensure correct values? 🤔 (Not sure it's a good idea for android though, maybe a stringdef would be better)

Reply:
Or, to avoid the possible error described below,
maybe you could have a sealed class like
```
sealed class TransactionTopic<T>
```

where `T` is the type of callback. This way, the `joinChannel` would take a `SocketTopic<T>` and a `T` instance? 🤔

Reply:
It would also be nice to be able to listen to **all** types of events at once, especially to write an Rx wrapper, where you could just wrap it in an observable, then just filter on the event type to subscribe to a particular event.
Not sure this is doable with this api, might be an issue if the dev has to manually register to all topics 🤔

Reply:
> It would also be nice to be able to listen to all types of events at once, especially to write an Rx wrapper, where you could just wrap it in an observable, then just filter on the event type to subscribe to a particular event.
Not sure this is doable with this api, might be an issue if the dev has to manually register to all topics 🤔

Do you mean put all events in the  **custom event** in one place?

Reply:
>Or, to avoid the possible error described below,
maybe you could have a sealed class like
```kotlin
sealed class TransactionTopic<T>
```
>where T is the type of callback. This way, the joinChannel would take a SocketTopic<T> and a T instance? 🤔

Now, the callback type need to be the same as the SocketTopic which is already declared its type in the response (see: `TransactionRequest`, `TransactionConsumption`). Please review this commit 👉 https://github.com/omisego/android-sdk/pull/24/commits/0d81e46ba19d712a895b2ecaea69d63a9d40c293